appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartData.java

```diff
@@ -0,0 +1,136 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.gwt.resources.client.ImageResource;
+import com.google.gwt.user.client.ui.InlineHTML;
+import org.pepstock.charba.client.data.Dataset;
+
+import java.util.List;
+
+public abstract class MockChartData extends MockVisibleComponent {
+    private static final String PROPERTY_COLOR = "Color";
+    private static final String PROPERTY_LABEL = "Label";
+    private static final String PROPERTY_PAIRS = "ElementsFromPairs";
+
+    // Temporary placeholder for the Chart Data image
+    private InlineHTML labelWidget;
+
+    protected MockChartModel chartModel;
+    protected MockChart chart;
+
+    private String currentElements = "";
+
+    /**
+     * Creates a new instance of a Mock Chart Data component.
+     *
+     * @param editor editor of source file the component belongs to
+     * @param type  type string of the component
+     * @param icon  icon of the component
+     */
+    MockChartData(SimpleEditor editor, String type, ImageResource icon) {
+        super(editor, type, icon);
+
+        labelWidget = new InlineHTML();
+        labelWidget.setStylePrimaryName("ode-SimpleMockComponent");
+        labelWidget.setText("LINE CHART DATA");
+        initComponent(labelWidget);
+    }
+
+    /**
+     * Adds the Mock Chart Data component to the specified Mock Chart component
+     * @param chart  Chart Mock component to add the data to
+     */
+    public void addToChart(MockChart chart) {
+        // Set widget to invisible
+        labelWidget.setVisible(false);
+        labelWidget.setWidth("0");
+        labelWidget.setHeight("0");
+
+        // Set references for Chart view and Chart model
+        this.chart = chart;
+        this.chartModel = chart.createChartModel();
+
+        // Set the properties to the Data Series
+        setDataSeriesProperties();
```

Comment:
One issue: When the MockChartData component is actually created, the properties are set automatically. However, the properties should only be set when the Data component is added to a Chart.

For now, I have made a method that updates all required properties. Is there a cleaner way, by perhaps forcing all the properties to be set only on this method call? (also see Line 101 if statement, which is part of the approach)

Reply:
For this method, see line 131.